### PR TITLE
refactor: SouthPane into functional component

### PR DESCRIPTION
**Pull request by AAfghahi**
### SUMMARY
Converted SouthPane into a functional component using TypeScript. 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Screen Shot 2021-03-17 at 4 41 09 PM

### CODE REVIEW

**Review by betodealmeida — APPROVED** [review]
Awesome!

File: superset-frontend/src/SqlLab/components/SouthPane.tsx [hunk 1]
```diff
@@ -83,33 +77,34 @@ const StyledPane = styled.div`
...
+  dataPreviewQueries,
+  actions,
+  activeSouthPaneTab = 'Results',
+  height,
+  databases,
+  offline = false,
+  displayLimit,
+}: SouthPanePropTypes) {
+  const innerTabContentHeight = height - TAB_HEIGHT;
+  const southPaneRef: any = useRef();
```
Comment (eschutho) [inline superset-frontend/src/SqlLab/components/SouthPane.tsx:92]: more specific type here?

File: superset-frontend/src/SqlLab/components/SouthPane.tsx [hunk 2]
```diff
@@ -83,33 +77,34 @@ const StyledPane = styled.div`
...
+  const renderResults = () => {
     let latestQuery;
-    const { props } = this;
-    if (props.editorQueries.length > 0) {
+    if (editorQueries.length > 0) {
       // get the latest query
-      latestQuery = props.editorQueries.find(
-        q => q.id === this.props.latestQueryId,
-      );
+      latestQuery = editorQueries.find(q => q.id === latestQueryId);
```
Comment (eschutho) [inline superset-frontend/src/SqlLab/components/SouthPane.tsx:107]: you can destructure the id for a little shorter syntax

File: superset-frontend/src/SqlLab/components/SouthPane.tsx [hunk 3]
```diff
@@ -157,55 +155,52 @@ export class SouthPane extends React.PureComponent {
...
     databases: sqlLab.databases,
     offline: sqlLab.offline,
   };
 }
 
-function mapDispatchToProps(dispatch) {
+function mapDispatchToProps(dispatch: Dispatch) {
   return {
-    actions: bindActionCreators(Actions, dispatch),
+    actions: bindActionCreators<any, any>(Actions, dispatch),
```
Comment (eschutho) [inline superset-frontend/src/SqlLab/components/SouthPane.tsx:202]: do you want to move the map state and dispatch to a new hooks file like the one that you created `superset-frontend/src/views/CRUD/data/database/state.ts`?

File: superset-frontend/src/SqlLab/components/SouthPane.tsx (hunk 1 above)
Comment (AAfghahi) [reply superset-frontend/src/SqlLab/components/SouthPane.tsx:92]: I forget, is there a preference between createRef and useRef?

**Comment by eschutho** [comment]
@hughhhh is going to finish up this review.. just the last three comments and do a visual test.. thank you!

File: superset-frontend/src/SqlLab/components/SouthPane.tsx (hunk 3 above)
Comment (hughhhh) [reply superset-frontend/src/SqlLab/components/SouthPane.tsx:202]: +1

**Review by hughhhh — APPROVED** [review]
🚢

**Pull request merged**